go/mysql/slave_status.go

```diff
@@ -23,7 +23,9 @@ import (
 
 // SlaveStatus holds replication information from SHOW SLAVE STATUS.
 type SlaveStatus struct {
-	Position            Position
+	Position Position
+	// RelayLogPosition will hold the Position the relay log is at. This is NOT supported in MariaDB, and will be nil for that flavor.
+	RelayLogPosition    *Position
```

Comment:
The `Position` struct is just a wrapper around a single interface value, so the zero value `Position{}` already represents a nil "pointer". We use `Position.IsZero()` to check for this, and `{Encode,Decode}Position()` already makes sure to represent this value in a way that preserves emptiness over the wire.

```suggestion
	// RelayLogPosition is the Position that the replica would be at if it
	// were to finish executing everything that's currently in its relay log.
	// However, some MySQL flavors don't expose this information,
	// in which case RelayLogPosition.IsZero() will be true.
	RelayLogPosition    Position
```

Reply:
Done.